src/main/java/aspects/LockAspect.java

```diff
@@ -40,7 +44,12 @@ public Object beforeLock(ProceedingJoinPoint joinPoint) throws Throwable {
             return joinPoint.proceed();
         } finally {
             if (lock != null) {
-                lock.unlock();
+                try {
+                    lock.unlock();
+                } catch (IllegalStateException e) {
+                    // Don't crash here, still return result
```

Comment:
not sure i have enough context to weigh in. but if this is basically a 5 minute timeout then I agree crashing and saying it timed out seems fine for now - especially if doing whatever is being done here results in some inconsistent state or undefined behavior.